heidrun/ia.rb

```diff
@@ -0,0 +1,238 @@
+  'frickartreferencelibrary' => 'Frick Art Reference Library'
+}
+
+# edm:dataProvider
+#   Meta.xml
+#     <contributor>;
+#     <collection>
+#       [code name that must be mapped to the full official name]
+#     </collection>
+#     NOTE: <collection> should appear first in the list of dataProviders
+data_provider_map = lambda do |record|
+  data_provider = record['collection'].map do |c|
+    if collections.key?(c.value)
+      collections[c.value]
+    else
+      c
+    end
+  end
+
+  data_provider.concat(record['contributor'])
+end
+
+# dcterms:identifier
+#   META.XML
+#     <identifier> may be the same thing;
+#     <call_number>Call number: "[value]"</call_number>;
+#     <datafield tag="035" ind1=" " ind2=" ">
+#       <subfield code="a">[value]</subfield>
+identifier_map = lambda do |record|
+  identifier = record['identifier']
+
+  identifier.concat(record['call_number'])
+
+  identifier.concat(record['marc']
+                      .field('record', 'datafield')
+                      .match_attribute(:tag, '035')
+                      .match_attribute(:ind1, ' ')
+                      .match_attribute(:ind2, ' ')
+                      .field('subfield')
+                      .match_attribute(:code, 'a'))
+end
+
+# dc:relation
+#   MARC.XML
+#     <datafield tag="440"> or
+#     <datafield tag="490"> or
+#     <datafield tag="800"> or
+#     <datafield tag="810"> or
+#     <datafield tag="830">;
+#     <datafield tag="785">
+#       [Add ". " between subfields; Do not include subfield w];
+#     <datafield tag="780">
+#       [Add ". " between subfields; Do not include subfield w]
+relation_map = lambda do |record|
+  relation = record['marc'].field('record', 'datafield')
+    .match_attribute(:tag) { |tag| %w(440 490 800 810 830).include?(tag) }
+
+  %w(785 780).each do |tag|
+    fld = record['marc'].field('record', 'datafield')
+                      .match_attribute(:tag, tag)
+                      .field('subfield')
+                      .match_attribute(:code) { |code| code != 'w' }
+                      .map(&:value).to_a.join('. ')
+
+    relation.concat([fld]) unless fld.blank?
+  end
+
+  relation
+end
+
+Krikri::Mapper.define(:ia, parser: Krikri::XmlParser) do
+  # edm:dataProvider
+  #   Meta.xml
+  #     <contributor>;
+  #     <collection>
+  #       [code name that must be mapped to the full official name]
+  #     </collection>
+  #     NOTE: <collection> should appear first in the list of dataProviders
+  dataProvider class: DPLA::MAP::Agent do
+    label record.map(&data_provider_map).flatten
+  end
+
+  # edm:isShownAt
+  #   META.XML <identifier-access>
+  isShownAt class: DPLA::MAP::WebResource do
+    uri record.field('identifier-access')
+  end
+
+  # edm:preview
+  #   We should pull these from Google Books API
+  #     if we can't unblink the gifs in IA
+  # preview class: DPLA::MAP::WebResource do
+  #   uri ???
+  # end
+  # TODO: what to do here? - JB
```

Comment:
cc @guegueng - previously we chose not to use `preview` since IA returns animated GIFs. We could add this here, but would need to add support for fetching the `files` metadata (see [comment on dpla/Krikri#208](https://github.com/dpla/KriKri/pull/208#discussion_r48173465))